src/modeling_library/dist/dist_dsl.jl

```diff
@@ -145,24 +143,20 @@ end
 macro dist(fnexpr)
   # First, pull out arguments and body
   fndef = splitdef(longdef(fnexpr))
-  #
-  # if !@capture(fnexpr, argexpr_ -> body_)
-  #     error("@dist expression must be of the form @dist (args...) -> body")
-  # end
-  # arguments = (typeof(argexpr) == Symbol) ? (argexpr,) : argexpr.args # [] #argexpr
-
   arguments = fndef[:args]
   body = fndef[:body]
 
   name_to_index = Dict{Symbol, Int8}()
+  name_to_type = Dict{Symbol, Type}()
   for (i, arg) in enumerate(arguments)
-      (argname, _, _, _) = splitarg(arg)
+      (argname, argtype, _, _) = splitarg(arg)
       name_to_index[argname] = i
+      name_to_type[argname] = eval(argtype)
   end
 
   function process_node(node)
       if typeof(node) == Symbol && haskey(name_to_index, node)
-          return :($(SimpleArg(name_to_index[node])))
+          return :($(SimpleArg{name_to_type[node]}(name_to_index[node])))
```

Comment:
Right -- I think I meant that you could do `:(SimpleArg{type}($(name_to_index[node])))` -- so that the SimpleArg is created in the generated code, rather than at compile-time. But if `__module__.eval` works, that's OK too!

Reply:
Mm right, I tried doing that then got errors, but I just realized it's because I didn't do `Gen.SimpleArg` instead of just `SimpleArg`!